src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -728,7 +728,7 @@ public ImmutableArray<T> Replace(T oldValue, T newValue)
         public ImmutableArray<T> Replace(T oldValue, T newValue, IEqualityComparer<T> equalityComparer)
         {
             var self = this;
-            int index = self.IndexOf(oldValue, equalityComparer);
+            int index = self.IndexOf(oldValue, 0, self.Length, equalityComparer);
```

Comment:
Thank you James, as I don't intend to submit this immediately, I'll make sure to merge your changes in. If you can get merged in next week or so, I don't expect that you'll have anything to do here. As with the other changes in immutable collections, I'd like to get an opinion from code maintainers whether they'd like me to split files up and do partial classes, or fun times with #if commands. 

@AArnott